docs/docs/user-guide/training-data-format.mdx

```diff
@@ -0,0 +1,659 @@
+---
+id: training-data-format
+sidebar_label: Training Data Format
+title: Training Data Format
+description: Description of the YAML format for training data
+---
+
+## Overview
+
+### YAML Format
+
+Rasa Open Source 2.0 uses [YAML](https://yaml.org/spec/1.2/spec.html) as
+a unified and extendable way to manage all training data,
+including NLU data, stories and rules. 
+
+With the YAML format, training data can be split over any number of YAML files,
+and every file can contain any kind of data. 
+The training data parser will read the top level keys in each file to decide
+what kind of data is in a section at training time.
+
+The [domain](../glossary#domain) uses
+the same YAML format as the training data and can also be split across
+multiple files or combined with training data in one file. The domain includes
+the definitions for [responses](../core/domains#responses) and [forms](../core/forms).
+See the [documentation for domains](../core/domains) to learn more.
+
+### High Level Structure
+
+Each file can contain one or more **keys** with corresponding training
+data. One file can contain multiple keys, as long as there is not more
+than one of a certain key in a single file. The available keys are:
+
+- `version`
+- `stories`
+- `nlu`
+- `rules`
+- `e2e_tests`
+
+All YAML training data files should specify the `version` key in order
+to be parsed correctly. Training data files with no `version` specified will be assumed to
+be in the format of the latest version of Rasa Open Source. Training data files with a version greater than is currently
+available for Rasa Open Source will be skipped.
+
+### Example
+
+Here's a short example which keeps all training data in a single file:
+
+```yaml
+version: "2.0"
+
+stories:
+- story: greet and faq
+  steps:
+  - intent: greet
+  - action: utter_greet
+  - intent: faq
+  - action: respond_faq
+
+rules:
+- rule: Greet user
+  steps:
+  - intent: greet
+  - action: utter_greet
+
+responses:
+  faq/language:
+  - text: |
+    I can only do English at the moment
+
+  utter_greet:
+  - text: |
+    Hey there!
+
+nlu:
+- intent: greet
+  examples: |
+  - Hey
+  - Hi
+  - hey there [Sara](name)
+
+- intent: faq/language
+  examples: |
+  - What language do you speak?
+  - Do you only handle english?
+
+e2e_tests:
+- user: |
+    hey
+  intent: greet
+- action: utter_greet
+- user: |
+    what language do you speak
+  intent: faq/language
+- action: respond_faq
+```
+
+
+## NLU Training Data
```

Comment:
should we rather call this something like "User Message Training Data"?

Reply:
I'd rather like to avoid the term NLU in most areas as the user likely doesn't know what it means

Reply:
even if we introduce NLU as a term in this section, we should avoid using these terms in the title of that section

Reply:
I'm not sure about this. If users are looking at the training data format (which will be not at the top the way it is now), I think they've already seen NLU and know what it is. We also talk about the NLU model, so it makes it clear what model it's feeding.

Reply:
It's true it should be defined first though